LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/DownstreamSender.cs

```diff
@@ -85,8 +85,8 @@ private string Message(DownlinkPktFwdMessage message, Region region)
                 writer.WriteNumber("RxDelay", message.LnsRxDelay);
                 writer.WriteNumber("RX1DR", dataRate);
                 writer.WriteNumber("RX1Freq", (ulong)(message.Txpk.Freq * 1e6));
-                writer.WriteNumber("RX2DR", region.GetDefaultRX2ReceiveWindow().DataRate);
-                writer.WriteNumber("RX2Freq", (ulong)region.GetDefaultRX2ReceiveWindow().Frequency);
+                writer.WriteNumber("RX2DR", region.GetDefaultRX2ReceiveWindow(message.DeviceJoinInfo).DataRate);
```

Comment:
We are getting the default RX2 receive window twice. I see that some of the regions have quite some logic to get to the window. Would it make sense to cache that information and only get it once?